### PR TITLE
[SPARK-11390] [SQL] Query plan with/without filterPushdown indistinguishable

### PR DESCRIPTION
**Pull request by zeocio**
…ishable

Propagate pushed filters to PhyicalRDD in DataSourceStrategy.apply

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/ExistingRDD.scala [hunk 1]
```diff
@@ -99,15 +99,16 @@ private[sql] case class PhysicalRDD(
...
 }
 
 private[sql] object PhysicalRDD {
   def createFromDataSource(
       output: Seq[Attribute],
       rdd: RDD[InternalRow],
-      relation: BaseRelation): PhysicalRDD = {
-    PhysicalRDD(output, rdd, relation.toString, relation.isInstanceOf[HadoopFsRelation])
+      relation: BaseRelation,
+      extraString: String = ""): PhysicalRDD = {
```
Comment (marmbrus) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ExistingRDD.scala]: Lets call it `extraInformation` in both places.

**Comment by marmbrus** [comment]
Can you fix the title / description so it doesn't truncate.  Maybe add a test too so this doesn't get lost in refactoring.

**Comment by zeocio** [comment]
@marmbrus  i was thinking about test cases, which sql/xxxSuite should this test case be added to?

**Comment by marmbrus** [comment]
maybe sql/core/src/test/scala/org/apache/spark/sql/execution/PlannerSuite.scala ?

**Comment by zeocio** [comment]
Changes to string formatting of Filter expr can break the test case. Did this recently change from "(key#27 = 15)" to "EqualTo(key,15)" ?

**Comment by zeocio** [comment]
@marmbrus  is the PR good to go?

**Comment by yhuai** [comment]
ok to test

**Comment by zeocio** [comment]
Please merge.

**Comment by marmbrus** [comment]
Thanks, merging to master and 1.6.

**Pull request closed without merging**